webapi/src/main/scala/org/knora/webapi/responders/v1/SipiResponderV1.scala

```diff
@@ -160,15 +160,22 @@ class SipiResponderV1 extends ResponderV1 {
             conversionResultResponse <- recoveredConversionResultFuture
 
             // get file type from Sipi response
-            responseAsMap: Map[String, JsValue] = conversionResultResponse.entity.asString.parseJson.asJsObject.fields.toMap
+            responseAsMap: Map[String, JsValue] = conversionResultResponse.entity.asString.parseJson.asJsObject.fields
 
             statusCode: Int = responseAsMap.getOrElse("status", throw SipiException(message = "Sipi did not return a status code")) match {
                 case JsNumber(ftype: BigDecimal) => ftype.toInt
                 case other => throw SipiException(message = s"Sipi did not return a correct status code, but ${other.toString()}")
             }
 
             // check if Sipi returned a status code != 0
-            _ = if (statusCode != 0) throw BadRequestException(s"Sipi returned a HTTP 200 status code, but an unsuccessful status code ${statusCode}")
+            _ = if (statusCode != 0) {
+                // Sipi returned an unsuccessful status code, get the error message
+                val sipiErrorMessage: String = responseAsMap.getOrElse("message", throw SipiException(message = "Sipi did not return an error message with key 'message' although status is != 0")) match {
+                    case JsString(msg) => msg
+                    case other => throw SipiException(message = s"Sipi did not return an error message of type string, but ${other.toString()}")
+                }
+                throw BadRequestException(s"Sipi returned an HTTP 200 status code, but an unsuccessful status code ${statusCode} with error message: '${sipiErrorMessage}'")
```

Comment:
Since the routes for Sipi are written in Lua, the user can freely define the JSON to be returned. We can just change the Lua scripts related to Knora: https://github.com/dhlab-basel/Sipi/tree/develop/scripts

For the status code, there is a method `sendStatus` which can be called in Lua:

```
-- if check returns false, the user's input is invalid
if not check then
    result = {
        status = 1,
        message = "Mimetypes and/or file extension are inconsistent."
    }

    jsonstr = server.table_to_json(result)

    server.sendStatus(400)
    server.print(jsonstr)

    return
end
```

I will change the Lua scripts using the pull request https://github.com/dhlab-basel/Sipi/pull/65:
- get rid of extra status code in JSON msg
- send an apt HTTP status code in case of an error using `server.sendStatus`